linux_os/guide/system/accounts/accounts-pam/locking_out_password_attempts/accounts_passwords_pam_faillock_deny/oval/shared.xml

```diff
@@ -1,222 +1,290 @@
   </ind:textfilecontent54_object>
 
+  <ind:textfilecontent54_test check="all" check_existence="none_exist" version="1"
+        id="test_accounts_passwords_pam_faillock_deny_password_pam_unix_auth"
+        comment="Only one occurrence of pam_unix.so is expected in auth section of password-auth">
+    <ind:object object_ref="object_accounts_passwords_pam_faillock_deny_password_pam_unix_auth"/>
+  </ind:textfilecontent54_test>
 
-  <!-- step 3 object -->
-  <ind:textfilecontent54_object id="object_accounts_passwords_pam_faillock_when_lines_skipped_system-auth" comment="Is pam_faillock not skipped?" version="1">
-    <ind:filepath>/etc/pam.d/system-auth</ind:filepath>
-    <ind:pattern operation="pattern match" var_ref="var_accounts_passwords_pam_faillock_preauth_default_lines_regex_system-auth"/>
+  <!-- Check common definition of pam_faillock.so in system-auth -->
+  <ind:textfilecontent54_object version="1"
+        id="object_accounts_passwords_pam_faillock_deny_system_pam_faillock_auth"
+        comment="Check common definition of pam_faillock.so in auth section of system-auth">
+    <ind:filepath operation="pattern match">^/etc/pam.d/system-auth$</ind:filepath>
+    <ind:pattern operation="pattern match"
+                 var_ref="var_accounts_passwords_pam_faillock_deny_pam_faillock_auth_regex"/>
     <ind:instance datatype="int" operation="equals">1</ind:instance>
   </ind:textfilecontent54_object>
 
-  <!-- step 5 -->
-  <local_variable id="var_accounts_passwords_pam_faillock_preauth_default_lines_regex_system-auth" datatype="string" version="1" comment="Regex containing skipped lines">
-    <concat>
-      <literal_component datatype="string">pam_unix(?:.*[\n](?:.*[\n]){</literal_component>
-      <object_component item_field="subexpression" object_ref="object_accounts_passwords_pam_faillock_lines_value_system-auth" />
-      <literal_component datatype="string">})(?:.*[\n])*auth.*pam_faillock.so[\s]+[^\n]*deny=([0-9]+)</literal_component>
-    </concat>
-  </local_variable>
-
-  <!-- end of system-auth handling, and beginning of password-auth handling -->
-  <!-- step 1 and 3 test -->
-  <ind:textfilecontent54_test id="test_accounts_passwords_pam_faillock_numeric_default_check_password-auth"
-  check="all" check_existence="all_exist"
-  comment="Checks if pam_faillock authfail is hit even if pam_unix skips lines by defaulting, and also authfail deny value" version="2">
-    <ind:object object_ref="object_accounts_passwords_pam_faillock_when_lines_skipped_password-auth" />
-    <ind:state state_ref="state_var_accounts_passwords_pam_faillock_deny_value_upper_bound" />
-    <ind:state state_ref="state_var_accounts_passwords_pam_faillock_deny_value_lower_bound" />
+  <ind:textfilecontent54_test check="all" check_existence="all_exist" version="1"
+        id="test_accounts_passwords_pam_faillock_deny_system_pam_faillock_auth"
+        comment="One and only one occurrence is expected in auth section of system-auth">
```

Comment:
To ensure that only one object exists, `check_existence` should be `only_one_exists`

Reply:
See https://github.com/ComplianceAsCode/content/pull/7929/files#discussion_r763053128

Reply:
I'm not sure what this link should point to..

Reply:
I'd still change the test to `check_existence="only_one_exists"` as that is the intention of the test.